tests/test_update_file_manager.py

```diff
@@ -71,3 +77,792 @@ def test_check_if_update_based_on_last_modified(database, mocker, mock_db):
     mock_requests.return_value.text = ""
 
     assert update_manager.check_if_update(url, float("-inf")) is True
+
+
+def test_check_if_update_local_file(database, mock_db):
+    """
+    Test if check_if_update_local_file() correctly detects if we should update
+    a local file based on the file hash
+    """
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    file_path = os.path.join(
+        os.getcwd(), "slips_files/ports_info/ports_used_by_specific_orgs.csv"
+    )
+    assert update_manager.check_if_update_local_file(file_path) is True
+    mock_db.get_TI_file_info.return_value = {
+        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+    }
+    assert update_manager.check_if_update_local_file(file_path) is True
+
+
+@pytest.mark.parametrize(
+    "mock_data, expected_feeds",
+    [
+        # Testcase1: Valid file with multiple feeds and tags.
+        (
+            "https://example.com/feed1.txt,"
+            "medium,tag1,tag2\nhttps://example.com/feed2.txt,high,tag3",
+            {
+                "https://example.com/feed1.txt": {
+                    "tags": ["tag1 tag2"],
+                    "threat_level": "medium",
+                },
+                "https://example.com/feed2.txt": {
+                    "tags": ["tag3"],
+                    "threat_level": "high",
+                },
+            },
+        ),
+        # Testcase2: File with comments and invalid threat level.
+        (
+            "# This is a comment\nhttps://example.com/feed3.txt,"
+            "invalid,tag4\nhttps://example.com/feed4.txt,low,tag5",
+            {
+                "https://example.com/feed3.txt": {
+                    "tags": ["tag4"],
+                    "threat_level": "low",
+                },
+                "https://example.com/feed4.txt": {
+                    "tags": ["tag5"],
+                    "threat_level": "low",
+                },
+            },
+        ),
+        # Testcase3: Empty file.
+        ("", {}),
+    ],
+)
+def test_get_feed_details(mocker, mock_db, mock_data, expected_feeds):
+    """Test get_feed_details with different file contents."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    m = mock_open(read_data=mock_data)
```

Comment:
alwayssss use descriptive variable names, always always alwayss. new developer will not know what m means. you may not know what m means 2 weeks from now :))